bool finish
---
int8 turn_angle